keps/sig-apps/4026-crojob-scheduled-timestamp-annotation/README.md

```diff
@@ -0,0 +1,289 @@
+
+### Test Plan
+
+- [X] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+
+##### Unit tests
+
+- `k8s.io/kubernetes/pkg/controller/cronjob`: `05/22/2023` - `96.2%`
+
+##### Integration tests
+
+- Unit tests will ensure the new annotation is correctly added to jobs.
+- The integration test should ensure the annotation is present when the feature is on and missing when off. It will also verify that the annotation is only added to jobs from newly created CronJobs, not existing workloads.
+
+##### e2e tests
+
+E2E tests will not provide any additional coverage that isn't already covered by unit + integration tests, since we are simply adding an annotation, so no e2e tests will be necessary for this change.
+
+### Graduation Criteria
+
+The feature will be released directly in Beta state since there is no benefit in having an alpha release, since we are simply adding a new annotation so there is very little risk.
+
+#### Beta
+
+- Feature implemented behind the `CronJobsScheduledAnnotation` feature gate.
+- Unit and integration tests passing.
+
+#### GA
+
+Fix any potentially reported bugs.
+
+### Upgrade / Downgrade Strategy
+
+No changes required to existing cluster to use this feature.
+
+### Version Skew Strategy
+
+N/A. This feature doesn't require coordination between control plane components,
+the changes to each controller are self-contained.
+
+## Production Readiness Review Questionnaire
+
+
+### Feature Enablement and Rollback
+
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+
+- [X] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: `CronJobCreationAnnotation`
+  - Components depending on the feature gate: `kube-controller-manager`
+- [ ] Other
+  - Describe the mechanism: N/A.
+  - Will enabling / disabling the feature require downtime of the control
+    plane? No
+  - Will enabling / disabling the feature require downtime or re-provisioning of a node? No
+
+###### Does enabling the feature change any default behavior?
+
+The jobs newly created by cronjob controller will contain a new annotation `CronJobsScheduledAnnotation`.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes. If the feature gate is disabled, the CronJob controller will not add the
+scheduled timestamp as an annotation.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The CronJob controller will begin adding the scheduled timestamp as an annotation to jobs created while the feature is enabled, and existing jobs will be unaffected.
+
+###### Are there any tests for feature enablement/disablement?
+
+We plan to add unit tests.
```

Comment:
Given the feature is somehow in-memory [we only add an information for newly created jobs], this isn't really needed (the fact that we enable/disable feature-gate in the middle of the test doesn't change anything).

Let's switch to:

Given the feature results in adding an annotation only to newly created objects, those tests won't really be different from the actual feature tests.